src/RoleObjectFactory.js

```diff
@@ -57,6 +57,8 @@ import ToolBarData from './RoleObjects/ToolBarData.js';
  * @param {number} [config.containerIndex] - An optional value for the layer index to add the DisplayObject
  * @param {String} [config.domIdPrefix] - Optional parameter for the prefix to use for the DOM id in the translated display object.  Defaults to 'acc_'
  * @param {Function} [config.onFocus] - Optional callback when the DisplayObject is brought into focus
+ * @param {Function} [config.onBlur] - Optional callback when the DisplayObject is gets blur
+ * @param {Function} [config.onValueChanged] -  Optional callback when the value changed for the DisplayObject
```

Comment:
CAM currently can generate about a dozen unique events between all the roles (documented in the "Events" part of the readme.md).  So, I would like something more generic that can cover all of them.  I was thinking that to the `config` object there could be an optional `events` object.  The `events` object would have fields named for the event and the value being the event listener.  While I haven't tested this code, it should allow for the event registration code to be along the lines of:
```
if (config.events) {
  _.forEach(config.events, (eventName, listener) => {
    displayObject.on(eventName, listener);
  });
}
```
That does mean the users of CAM can bind event listeners for a role that will never fire that event or an event that CAM will never fire (but those cases are already handled by documenting roles and their events in the readme.md), but it allows for consistent event registration regardless of role.  This also allows for the previous code for onFocus and onKeyboardClick to be removed since those should use the `events` object going forward (since this is part of the 1.0 release, API breaking changes are okay).

Reply:
updated